src/app/modules/billing/billing.controller.ts

```diff
@@ -0,0 +1,76 @@
+import { RequestHandler } from 'express'
+import { ParamsDictionary } from 'express-serve-static-core'
+import { StatusCodes } from 'http-status-codes'
+import moment from 'moment-timezone'
+
+import { createLoggerWithLabel } from '../../../config/logger'
+import { createReqMeta } from '../../utils/request'
+import { isUserInSession } from '../auth/auth.utils'
+
+import { BillingFactory } from './billing.factory'
+
+const logger = createLoggerWithLabel(module)
+
+/**
+ * Handler for GET /billing endpoint.
+ * @return 200 with login statistics when query is valid
+ * @return 401 when request does not contain a user session
+ * @return 500 when error occurs whilst querying database
+ */
+export const handleGetBillInfo: RequestHandler<
+  ParamsDictionary,
+  unknown,
+  unknown,
+  {
+    esrvcId: string
+    yr: string
+    mth: string
+  }
+> = async (req, res) => {
+  // Restricted route.
+  if (!isUserInSession(req)) {
+    return res.status(StatusCodes.UNAUTHORIZED).json('User is unauthorized.')
+  }
+
+  const { esrvcId, mth, yr } = req.query
+
+  const startOfMonth = moment
+    .tz([parseInt(yr), parseInt(mth)], 'Asia/Singapore')
+    .startOf('month')
+  const endOfMonth = moment(startOfMonth).endOf('month')
+
+  const loginStatsResult = await BillingFactory.getSpLoginStats(
+    esrvcId,
+    startOfMonth.toDate(),
+    endOfMonth.toDate(),
+  )
+
+  if (loginStatsResult.isErr()) {
```

Comment:
the original code has a `loginStatsResult.isOk() && !loginStatsResult.value` case as well, is there any chance of that happening?

Reply:
no, `aggregate` always returns an array; but the array can be empty